### PR TITLE
Fix: Increase MaxMemory param to use 1/4 of total memory.

### PR DESCRIPTION
**Pull request by ajnavarro**
Per discussion here: https://github.com/ipfs/kubo/pull/9443#issuecomment-1334831010
We are going to set the default max memory to 1/2 of the total memory instead of 1/8 to improve the default total inbound connections number, trying to avoid CM errors.

Signed-off-by: Antonio Navarro Perez <antnavper@gmail.com>

### CODE REVIEW

**Review by BigLep — APPROVED** [review]
@ajnavarro : looks good to me.  I think it's also useful to give context in the PR where this is coming from: https://github.com/ipfs/kubo/pull/9443#issuecomment-1334831010

This should also be added to https://github.com/ipfs/kubo/issues/9442

**Comment by ajnavarro** [comment]
@BigLep added a comment on the PR description.

**Comment by BigLep** [comment]
Per 2022-12-06 maintainer conversation, we're going to go from half to a quarter of system memory.

File: core/node/libp2p/rcmgr_defaults.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ var noLimitIncrease = rcmgr.BaseLimitIncrease{
 // The defaults follow the documentation in docs/config.md.
 // Any changes in the logic here should be reflected there.
 func createDefaultLimitConfig(cfg config.SwarmConfig) (rcmgr.LimitConfig, error) {
-	maxMemoryDefaultString := humanize.Bytes(uint64(memory.TotalMemory()) / 8)
+	maxMemoryDefaultString := humanize.Bytes(uint64(memory.TotalMemory()) / 2)
```
Comment (BigLep) [inline core/node/libp2p/rcmgr_defaults.go:50]:
```suggestion
	maxMemoryDefaultString := humanize.Bytes(uint64(memory.TotalMemory()) / 4)
```

File: docs/config.md [hunk 2]
```diff
@@ -1865,7 +1869,7 @@ This value is also used to scale the limit on various resources at various scope
 when the default limits (discuseed above) are used.
 For example, increasing this value will increase the default limit for incoming connections.
 
-Default: `[TOTAL_SYSTEM_MEMORY]/8`
+Default: `[TOTAL_SYSTEM_MEMORY]/2`
```
Comment (BigLep) [inline docs/config.md:1872]:
```suggestion
Default: `[TOTAL_SYSTEM_MEMORY]/4`
```

**Review by BigLep — APPROVED** [review]

**Pull request merged**